### PR TITLE
What-if UI elements

### PR DESCRIPTION
**Pull request by szabinst**
WhatIf UI

refs: MBL-17303
affects: Student, Parent
release note: none
test plan: Check figma

## Notes
- Using `List` now to access `swipeActions` 
- Because of this you'll see that there's a space before section headers so I decided to use a clear background for the custom view and let the OS color the header. Otherwise there would be a weird transition with colors when the stretched out header scales down in size when scrolling. 

## Screenshots

<table>
<tr><th>Toggle</th><th>Editor</th></tr>
<tr>
<td></td>
<td></td>
</tr>
<tr><th>Revert</th><th>Swipe action</th></tr>
<tr>
<td></td>
<td></td>
</tr>
</table>

## Checklist

- [ ] Follow-up e2e test ticket created
- [ ] A11y checked
- [x] Tested on phone
- [x] Tested on tablet
- [x] Tested in dark mode
- [x] Tested in light mode
- [ ] Approve from product

### CODE REVIEW

File: Core/Core/Grades/View/GradeListView.swift [hunk 1]
```diff
@@ -377,31 +401,120 @@ public struct GradeListView: View, ScreenViewTrackable {
...
-                        }
-                        .buttonStyle(ContextButton(contextColor: courseColor))
-                        Divider()
+                isScoreEditorPresented.toggle()
+            }
+        }
+        .listRowInsets(EdgeInsets())
+        .iOS16RemoveListRowSeparatorLeadingInset()
+        .swipeActions(edge: .trailing) { revertWhatIfScoreSwipeButton }
+//        .buttonStyle(ContextButton(contextColor: courseColor))
```
Comment (szabinst) [inline Core/Core/Grades/View/GradeListView.swift:458]: buttonStyle doesn't play nicely with list row. If buttonStyle is added, additional buttons can't be pressed.

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Student, Parent

[MBL-17303](https://instructure.atlassian.net/browse/MBL-17303)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ceee4a3ee38bc04e799d5cf294c67e44149bae8b
</p>

**Comment by inst-danger** [comment]
<details><summary>Parent Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Review by vargaat — CHANGES_REQUESTED** [review]
I tested on an iPad with iOS 16.7.1, here are my findings:

1. The what-if text box is too small. I entered 123 but only 2 digits are displayed.
2. The baseline of the text fields and their labels are not aligned vertically.
IMG_0314
3. The edit grade icon is always displayed, even when I have the what-if score disabled.
IMG_0313
4. The dialog doesn't seem to respect dynamic text size settings.
IMG_0315

**Comment by ndrsszsz** [comment]
If I turn the phone to landscape then the list with the assignments and grades disappear like this:

IMG_0003

**Comment by szabinst** [comment]
@ndrsszsz I fixed it, please check again.

**Review by ndrsszsz — APPROVED** [review]
QA +1

**Review by vargaat — APPROVED** [review]
QA+1

**Comment by vargaat** [comment]
The swipe to revert gesture should be disabled if the what if score toggle is off in the next PR.

**Comment by szabinst** [comment]
@vargaat Good catch. Will do that.

**Pull request merged**